src/napari/layers/utils/stack_utils.py

```diff
@@ -251,15 +253,22 @@ def stack_to_images(stack: Image, axis: int, **kwargs) -> list[Image]:
 
 
 def split_rgb(stack: Image, with_alpha=False) -> list[Image]:
-    """Variant of stack_to_images that splits an RGB with predefined cmap."""
+    """Split RGB image into separate channel images while preserving affine transforms."""
     if not stack.rgb:
         raise ValueError(
             trans._('Image must be RGB to use split_rgb', deferred=True)
         )
-    # representing alpha channel as gray
-    images = stack_to_images(
-        stack, -1, colormap=('red', 'green', 'blue', 'gray')
-    )
+
+    data, meta, _ = stack.as_layer_data_tuple()
+
+    meta['colormap'] = ('red', 'green', 'blue', 'gray')
+    meta.pop('rgb')
```

Comment:
```suggestion
    meta['rgb'] = False
```

I prefer explicit, but feel free to ignore.